src/renderer/components/battle/BotParticipant.vue

```diff
@@ -59,6 +63,13 @@ function kickBot() {
     props.battle.removeBot(props.bot);
 }
 
+// Duplicates this bot and its settings and gives it a new player id.
+function duplicateBot() {
+    let duplicatedBot = JSON.parse(JSON.stringify(props.bot));
```

Comment:
[`structuredClone`](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) should be cleaner in what you are trying to do here vs pasing object via JSON serialization. It also should be better for typing.

Also change `let` to `const`.

Reply:
Thanks! Adjustments made in a877be / 959a9fc, alongside a fix for options not being set by some sort of Id.